Lib/pathlib.py

```diff
@@ -68,7 +68,12 @@ def _make_selector(pattern_parts, flavour, case_sensitive):
     if not pat:
         return _TerminatingSelector()
     if pat == '**':
-        cls = _RecursiveWildcardSelector
+        while child_parts and child_parts[0] == '**':
+            child_parts = child_parts[1:]
```

Comment:
This feels a little icky because we're repeatedly copying the list (I assume it's a list), so this is quadratic. I suppose you wouldn't write a glob with many leading `**` segments, though.

An alternative could be to store the index of the first non-`**` part in a variable, and then loop over the list to find the right value.

Reply:
It's a tuple! It needs to be hashable because it's passed back into `_make_selector` later. In fact, this code is only doing what subsequent calls to `_make_selector` would have done in the first couple of lines.

I'll try to count the number of leading `**` and slice once.